VideoLocker/src/main/java/org/edx/mobile/view/CourseDashboardFragment.java

```diff
@@ -159,6 +172,44 @@ public void onActivityCreated(Bundle savedInstanceState) {
         courseTextName.setText(courseData.getCourse().getName());
         CourseEntry course = courseData.getCourse();
         courseTextDetails.setText(course.getDescription(getActivity(), true));
+
+        if (environment.getConfig().isShareCourseEnabled()) {
+            shareButton.setVisibility(headerImageView.VISIBLE);
+            shareButton.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    openShareMenu();
+                }
+            });
+        }
+    }
+
+    /**
+     * Creates a dropdown menu with appropriate apps when the share button is clicked.
+     */
+    private void openShareMenu() {
+        ShareUtils.showShareMenu(
+                ShareUtils.newShareIntent(courseData.getCertificateURL()),
```

Comment:
you copy-pasta'd my certificate URL. It wouldn't really matter except that certificate URL might be null, which might do bad things... You should use your intended shareText here instead.